interfaces/core.go

```diff
@@ -157,6 +157,18 @@ type Interface interface {
 	AutoConnect(plug *Plug, slot *Slot) bool
 }
 
+// Recorder describes interactions between backends and interfaces.
+type Recorder interface {
+	// RecordPermanentSlot records side-effects of having a slot.
+	RecordPermanentSlot(iface interface{}, slot *Slot) error
```

Comment:
There are some methods that every interface must have, right?  Would be better to use such an interface here instead of just `interface{}`.